### PR TITLE
Chore: Migrate SidebarView to Typescript

### PR DESCRIPTION
**Pull request by reinaldonetof**
## How to test or reproduce

- The sidebar should be work normally. Besides the sidebar the Button to Admin Panel must be working properly, I changed a small part of the algorithm to show this button.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/Status/Status.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { STATUS_COLORS } from '../../constants/colors';
 interface IStatus {
 	status: string;
 	size: number;
-	style: any;
+	style?: any;
```
Comment (AlexAlexandre) [inline app/containers/Status/Status.tsx:9]: let's change this `any` to `StyleSheet`;

File: app/views/SidebarView/index.tsx [hunk 2]
```diff
@@ -242,7 +250,7 @@ class Sidebar extends Component {
 					]}
 					{...scrollPersistTaps}>
 					<TouchableWithoutFeedback onPress={this.onPressUser} testID='sidebar-close-drawer'>
-						<View style={styles.header} theme={theme}>
+						<View style={styles.header}>
```
Comment (AlexAlexandre) [inline app/views/SidebarView/index.tsx:253]: why did you remove this theme ?

File: app/views/SidebarView/index.tsx [hunk 3]
```diff
@@ -127,17 +138,14 @@ class Sidebar extends Component {
...
 
 		if (roles) {
-			isAdmin = allPermissions.reduce((result, permission) => {
-				if (permission) {
-					return result || permission.some(r => roles.indexOf(r) !== -1);
-				}
-				return result;
-			}, false);
+			// Creating a new array with booleans values, than the every check if it is a truthy array
+			isAdmin = allPermissions.map(permission => permission.some(r => roles.indexOf(r) !== -1)).every(Boolean);
```
Comment (AlexAlexandre) [inline app/views/SidebarView/index.tsx:142]: when we don't have admin permissions we get this error.


Screen Shot 2021-11-02 at 00 15 15


the const `allPermissions` is like this: `[
  undefined,
  undefined,
  undefined,
  undefined
]`

**Review by AlexAlexandre — CHANGES_REQUESTED** [review]

File: app/views/SidebarView/index.tsx (hunk 3 above)
Comment (reinaldonetof) [reply app/views/SidebarView/index.tsx:142]: Nice, thanks, I didn't realize this!

**Review by AlexAlexandre — APPROVED** [review]
LGTM ;)

**Review by diegolmello — APPROVED** [review]
